homeassistant/components/remote/__init__.py

```diff
@@ -0,0 +1,151 @@
+
+
+def turn_on(hass, activity=None, entity_id=None):
+    """Turn all or specified remote on."""
+    data = {ATTR_ACTIVITY: activity}
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+def turn_off(hass, entity_id=None):
+    """Turn all or specified remote off."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_TURN_OFF, data)
+
+
+def send_command(hass, device=None, command=None, entity_id=None):
+    """Send a command to a device."""
+    data = {ATTR_DEVICE: str(device), ATTR_COMMAND: command}
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+    hass.services.call(DOMAIN, SERVICE_SEND_COMMAND, data)
+
+
+def sync(hass, entity_id=None):
+    """Sync remote device."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_SYNC, data)
+
+
+def setup(hass, config):
+    """Track states and offer events for remotes."""
+    component = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL, GROUP_NAME_ALL_REMOTES)
+    component.setup(config)
+
+    def handle_remote_service(service):
+        """Handle calls to the remote services."""
+        target_remotes = component.extract_from_service(service)
+
+        activity_id = service.data.get(ATTR_ACTIVITY, ATTR_DEFAULT)
+        device = str(service.data.get(ATTR_DEVICE, ATTR_DEFAULT))
+        command = str(service.data.get(ATTR_COMMAND, ATTR_DEFAULT))
+
+        for remote in target_remotes:
+            if service.service == SERVICE_TURN_ON:
+                remote.turn_on(activity=activity_id)
+            elif service.service == SERVICE_SEND_COMMAND:
+                remote.send_command(device=device, command=command)
+            elif service.service == SERVICE_SYNC:
+                remote.sync()
+            else:
+                remote.turn_off()
+
+            if remote.should_poll:
+                remote.update_ha_state(True)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+    hass.services.register(DOMAIN, SERVICE_TURN_OFF, handle_remote_service,
+                           descriptions.get(SERVICE_TURN_OFF),
+                           schema=REMOTE_SERVICE_SCHEMA)
+    hass.services.register(DOMAIN, SERVICE_TURN_ON, handle_remote_service,
+                           descriptions.get(SERVICE_TURN_ON),
+                           schema=REMOTE_SERVICE_SCHEMA)
+    hass.services.register(DOMAIN, SERVICE_SEND_COMMAND, handle_remote_service,
+                           descriptions.get(SERVICE_SEND_COMMAND),
+                           schema=REMOTE_SERVICE_SCHEMA)
+    hass.services.register(DOMAIN, SERVICE_SYNC, handle_remote_service,
+                           descriptions.get(SERVICE_SYNC),
+                           schema=REMOTE_SERVICE_SCHEMA)
+
+    return True
+
+
+class RemoteDevice(ToggleEntity):
```

Comment:
Is current_activity a thing ? If so, would it make sense to include that as a property and maybe even make it the state of a remote ?